### PR TITLE
feat: Add project id detection to the Google exporter

### PR DESCRIPTION
**Pull request by jmwilliams89**
### Proposed Change
- This change adds project id extraction to the Google exporter. The upstream exporter already supports detecting project id based on automatic credentials. This change achieves parity when the `credentials` or `credentials_file` field is configured by the user.
- The exporter makes a best attempt at extracting the project id. If it fails to find a project id in the specified credentials, it will log the error, but it will not cause the exporter to explicitly fail.
- The exporter will not attempt to extract the project id for the user if they have already specified a project id in their config.

##### Checklist
- [x] Changes are tested
- [x] CI has passed

### CODE REVIEW

File: exporter/googlecloudexporter/config.go [hunk 1]
```diff
@@ -69,6 +73,51 @@ func (c *Config) getClientOptions() []option.ClientOption {
...
+	if !ok {
+		return errors.New("project id is not a string")
+	}
+
+	c.GCPConfig.ProjectID = strValue
+	return nil
+}
+
+func (c *Config) updateProjectFromFile(fileName string) error {
+	jsonBytes, err := ioutil.ReadFile(fileName)
```
Comment (BinaryFissionGames) [inline exporter/googlecloudexporter/config.go:113]: Nit: ioutil is deprecated, the functions were split between the os and io packages
```suggestion
	jsonBytes, err := os.ReadFile(fileName)
```

File: exporter/googlecloudexporter/config.go [hunk 2]
```diff
@@ -69,6 +73,51 @@ func (c *Config) getClientOptions() []option.ClientOption {
...
+	if !ok {
+		return errors.New("project id is not a string")
+	}
+
+	c.GCPConfig.ProjectID = strValue
+	return nil
+}
+
+func (c *Config) updateProjectFromFile(fileName string) error {
+	jsonBytes, err := os.ReadFile(fileName)
```
Comment (BinaryFissionGames) [inline exporter/googlecloudexporter/config.go:113]: Looks like gosec is complaining about this, I think the filename just needs to be passed through filepath.Clean first
https://securego.io/docs/rules/g304.html

**Review by schmikei — APPROVED** [review]
I was curious about if the collector was deployed to GCP if there was any funky things that could happen with the default credentials loading. 

Its an edge case but something I wished to validate the behavior of, so I spun up a vm instance and I found that the override does in fact get used with precedence over the default application creds. 

TLDR; did some more extraneous validation, seems to be expected behavior 🔥
